public/static/js/spawned_hosts.js

```diff
@@ -92,6 +70,68 @@ mciModule.controller('SpawnedHostsCtrl', ['$scope','$window', '$timeout', 'mciSp
     }
 
 
+
+    $scope.updateSpawnedHosts = function() {
+        mciSpawnRestService.getSpawnedHosts(
+            'hosts', {}, {
+                success: function(resp) {
+                    var hosts = resp.data;
+                    _.each(hosts, function(host) {
```

Comment:
You're looping over hosts in a loop over hosts here

Reply:
Well, I'm looping over our existing hosts in a loop over the newly given hosts, in order to populate the existing host with the new data (I don't want to assume the ordering is the same). That being said, there are never going to be a crazy number of spawn hosts so I think the complexity will be fine.